IMS.BL/Application/Program.cs

```diff
@@ -88,12 +94,9 @@ public static void Main(string[] args)
             }
             catch (Exception e)
             {
-                if (e.GetType() == typeof(NullReferenceException))
-                {
-                    Console.WriteLine("There are no products with the provided data!");
-                }
-                
-                Console.WriteLine("The data you entered is not valid, please try again and provide a valid data!");
+                Console.WriteLine(e.GetType() == typeof(NullReferenceException)
+                    ? "There are no products with the provided data!"
+                    : "The data you entered is not valid, please try again and provide a valid data!");
                 Environment.Exit(1);
```

Comment:
you should handle each case by itself. not relaying on a general catch then using reflection to decide its type 
you should do more custom exception if needed and then catch each one of them 
```
catch(NullReferenceException e){
  handle exception
},catch(MyCustomeException e){
  handle exception
}.catch(Exception){
  generic message
}
```